strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+- _comeInWithDoorStuckSetup_: This indicates that Samus must enter the room in a way that allows getting stuck in the door as it closes.
+
+Each of these properties is described in more detail below.
+
+### Come In Running
+
+An `comeInRunning` entrance condition represents the need for Samus to be able to run into the room with speed in a certain range. It has the following properties:
+* _speedBooster_: If true, then Speed Booster must be used while running into the room. If false, then Speed Booster must not be used. If "any", then Speed Booster may or may not be used.
+* _minTiles_: The minimum horizontal speed that will satisfy the condition, measured in effective runway tiles with dash held.
+* _maxTiles_: The maximum horizontal speed that will satisfy the condition, measured in effective runway tiles with dash held.
+
+A `comeInRunning` condition can be satisfied only by a matching strat on the other side of the door with `leaveWithRunway` exit condition: a match is valid if all of the following are true:
+- The effective runway length of the `leaveWithRunway` is at least as long as the `minTiles` in the `comeInRunning` condition.
+
+Where applicable, a `comeInRunning` condition also includes implicit requirements for actions to be performed in the previous room, which are effectively prepended to the start of the strat's `requires` (or equivalently but more properly, onto the end of the `requires` of the `leaveWithRunway` strat in the other room):
+- If `speedBooster` is true, then there is an implicit "SpeedBooster" item requirement.
+- If the previous room is heated, then `heatFrames` are required based on the time needed to perform the run. 
+- If the previous door environment is water, then `Gravity` is required.
+
+Heat frames may be calculated based on the following table of approximations (which include almost no lenience):
+
+| Action                                                                  | Heat frames             |
+| ----------------------------------------------------------------------  | ----------------------- |
+| Run from a standstill in one direction, without Speedbooster, 0-6 tiles | `9 + 4 * tiles`         |
+| Run from a standstill in one direction, without Speedbooster, 7+ tiles  | `13 + 64 / 19 * tiles`  |
+| Run from a standstill in one direction, with Speedbooster, 0-6 tiles    | `9 + 4 * tiles`         |
+| Run from a standstill in one direction, with Speedbooster, 7-16 tiles   | `15 + 3 * tiles`        |
+| Run from a standstill in one direction, with Speedbooster, 17-42 tiles  | `32 + 2 * tiles`        |
+| Run from a standstill in one direction, with Speedbooster, 43+ tiles    | `47 + 64 / 39 * tiles`  |
+| Turn around                                                             | `8`                     |
+| Position using moonwalk (and shoot open the door)                       | `12`                    |
```

Comment:
what if the runway is longer than the max tiles? should these still be added in as if Samus were not running?

Reply:
In heated rooms at least, we will need to provide the accurate runway length no matter how long.